wasm/jit/jit_amd64.go

```diff
@@ -766,7 +766,128 @@ func (c *amd64Compiler) compileSub(o *wazeroir.OperationSub) error {
 		return err
 	}
 
-	// x1 += x2.
+	// x1 -= x2.
+	prog := c.newProg()
+	prog.From.Type = obj.TYPE_REG
+	prog.From.Reg = x2.register
+	prog.To.Type = obj.TYPE_REG
+	prog.To.Reg = x1.register
+	prog.As = instruction
+	c.addInstruction(prog)
+
+	// We no longer need x2 register after ADD operation here,
+	// so we release it.
+	c.locationStack.releaseRegister(x2)
+	return nil
+}
+
+func (c *amd64Compiler) compileMul(o *wazeroir.OperationMul) (err error) {
```

Comment:
```suggestion
// compileMul blah blah and saves any prior value on the DX register to the stack
func (c *amd64Compiler) compileMul(o *wazeroir.OperationMul) (err error) {
```

Reply:
generally I think it would be neat to start doing docs on the compileMul which minimally say at least the stack post conditions. I know preconditions are very complicated, but where simple enough those also.

Ex.

```
// compileMul adds instructions to multiply two operands which may be on the stack or registers.
// After execution, the following results will be on the stack:
// * stackPointer-1: result of the multiplication
// * stackPointer-2: the prior value of the DX register
```

Reply:
noted!